File: CSharp/SpecStudioExamplesTestsCSharp/common/NewPostTyped.cs
namespace SpecStudioExamplesTestsCSharp.common
{
using System;
using System.Buffers;
using System.Collections.Generic;
using System.Globalization;
using System.Text;
using System.Text.Json;
using production;

    public class NewPostTyped
    {
        public string title;
        public string body;
        public int userId;

        public NewPostTyped(string title, string body, int userId)
        {
            this.title = title;
            this.body = body;
            this.userId = userId;
        }

        public NewPostString ToNewPostString()
        {
            return new NewPostString(
                Json.ToText(this.title),
                Json.ToText(this.body),
                Json.ToText(this.userId)
            );
        }

        public static List<NewPostString> ToStringList(List<NewPostTyped> list)
        {
            var result = new List<NewPostString>();
            foreach (var t in list) result.Add(t.ToNewPostString());
            return result;
        }

        public static List<NewPostTyped> FromStringList(List<NewPostString> list)
        {
            var result = new List<NewPostTyped>();
            foreach (var s in list) result.Add(s.ToNewPostTyped());
            return result;
        }

        public void WriteJson(Utf8JsonWriter w)
        {
            w.WriteStartObject();
            w.WriteString("title", this.title);
            w.WriteString("body", this.body);
            w.WriteNumber("userId", this.userId);
            w.WriteEndObject();
        }

        public string ToJSON()
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer)) { WriteJson(w); }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static NewPostTyped FromJsonElement(JsonElement m)
        {
            return new NewPostTyped(
                Json.AsString(Json.Require(m, "title"), "title"),
                Json.AsString(Json.Require(m, "body"), "body"),
                Json.AsInt(Json.Require(m, "userId"), "userId")
            );
        }

        public static NewPostTyped FromJSON(string json)
        {
            return FromJsonElement(Json.Parse(json));
        }

        public static string ToJSONList(List<NewPostTyped> list)
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer))
            {
                w.WriteStartArray();
                foreach (var item in list) item.WriteJson(w);
                w.WriteEndArray();
            }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static List<NewPostTyped> FromJSONList(string json)
        {
            var result = new List<NewPostTyped>();
            var root = Json.Parse(json);
            Json.RequireArray(root, "NewPostTyped");
            foreach (var e in root.EnumerateArray()) result.Add(FromJsonElement(e));
            return result;
        }

        public override string ToString()
        {
            return $"title={title}, body={body}, userId={userId}";
        }

        public override bool Equals(object? obj)
        {
            if (obj is not NewPostTyped other) return false;
            return object.Equals(this.title, other.title)
                && object.Equals(this.body, other.body)
                && object.Equals(this.userId, other.userId);
        }

        public override int GetHashCode()
        {
            var h = new System.HashCode();
            h.Add(this.title);
            h.Add(this.body);
            h.Add(this.userId);
            return h.ToHashCode();
        }
    }
}

File: Java/SpectableJavaExamplesTest/src/test/java/spectable/common/NewPostTyped.java
package spectable.common;

import java.util.ArrayList;
import java.util.List;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Objects;
import production.*;
import records.*;
import calculator.*;

public class NewPostTyped {
    public String title;
    public String body;
    public int userId;

    public NewPostTyped(String title, String body, int userId) {
        this.title = title;
        this.body = body;
        this.userId = userId;
    }

    public NewPostTyped(NewPostString s) {
        this.title = s.title;
        this.body = s.body;
        this.userId = Integer.parseInt(s.userId);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof NewPostTyped)) return false;
        NewPostTyped that = (NewPostTyped) o;
        return Objects.equals(title, that.title)
            && Objects.equals(body, that.body)
            && Objects.equals(userId, that.userId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(title, body, userId);
    }

    public NewPostString toNewPostString() {
        return new NewPostString(String.valueOf(title), String.valueOf(body), String.valueOf(userId));
    }

    public static List<NewPostTyped> fromStringList(List<NewPostString> list) {
        List<NewPostTyped> result = new ArrayList<>();
        for (NewPostString s : list) result.add(new NewPostTyped(s));
        return result;
    }

    public static List<NewPostString> toStringList(List<NewPostTyped> list) {
        List<NewPostString> result = new ArrayList<>();
        for (NewPostTyped t : list) result.add(t.toNewPostString());
        return result;
    }

    /** Internal plumbing for {@link Json}; use toJSON() for JSON text. */
    public Map<String, Object> toJsonValue() {
        Map<String, Object> m = new LinkedHashMap<String, Object>();
        m.put("title", title);
        m.put("body", body);
        m.put("userId", userId);
        return m;
    }

    public String toJSON() { return Json.write(toJsonValue()); }

    /** Internal plumbing for {@link Json}; use fromJSON(String) for JSON text. */
    public static NewPostTyped fromJsonValue(Map<String, Object> m) {
        if (m == null) return null;
        return new NewPostTyped(
            Json.asString(Json.require(m, "title"), "title"),
            Json.asString(Json.require(m, "body"), "body"),
            Json.asInt(Json.require(m, "userId"), "userId"));
    }

    public static NewPostTyped fromJSON(String json) {
        return fromJsonValue(Json.parseObject(json));
    }

    public static String toJSONList(List<NewPostTyped> list) {
        List<Object> arr = new ArrayList<Object>();
        if (list != null)
            for (NewPostTyped item : list) arr.add(item == null ? null : item.toJsonValue());
        return Json.write(arr);
    }

    public static List<NewPostTyped> fromJSONList(String json) {
        List<NewPostTyped> result = new ArrayList<NewPostTyped>();
        for (Object e : Json.parseArray(json))
            result.add(fromJsonValue(Json.asObject(e, "NewPostTyped")));
        return result;
    }
}
